### PR TITLE
[operator] Deploy `BackupBucket`/`DNSRecord` extension resources

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/9635

**Special notes for your reviewer**:

**Release note**:

```other operator
Enhance the gardener-operator to deploy dnsrecords and backupbucket for the virtual garden resource.
```

### CODE REVIEW

File: pkg/operator/controller/garden/garden/components.go [hunk 1]
```diff
@@ -828,6 +831,14 @@ func getAPIServerDomains(domains []string) []string {
 	return apiServerDomains
 }
 
+func getIngressWildcardDomains(domains []string) []string {
+	wildcardDomains := make([]string, 0, len(domains))
+	for _, domain := range domains {
+		wildcardDomains = append(wildcardDomains, "*."+domain)
```
Comment (timebertt) [inline pkg/operator/controller/garden/garden/components.go:837]: @MichaelEischer and I just stumbled upon this in our gardener-operator setup:
When using a wildcard record for `*.<ingress-domain>` and cert-manager with DNS01 ACME, the cert-manager's TXT records for `_acme-challenge.{dashboard,identity,...}.<ingress-domain>` will cause the concrete records under the ingress domain (`{dashboard,identity,...}.<ingress-domain>`) to be unresolvable.

From https://en.wikipedia.org/wiki/Wildcard_DNS_record
> [...] the wildcard is matched only when a domain does not exist, not just when there are no matching records of the type that has been queried for.

However, this seems to depend on the concrete DNS implementation:
> The exact rules for when a wildcard will match are specified in [RFC](https://en.wikipedia.org/wiki/RFC_(identifier)) [1034](https://datatracker.ietf.org/doc/html/rfc1034), but the rules are neither intuitive nor clearly specified. This has resulted in incompatible implementations and unexpected results when they are used.

In short: we suggest not using wildcard records but concrete records for the ingresses gardener-operator deploys. This will make things compatible with using cert-manager's DNS01 ACME.
BTW: due to the nature of ACME challenges (at least for [Let's Encrypt](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge)), this also applies to setups where gardener's [cert-management](https://github.com/gardener/cert-management/issues/173) is used instead of cert-manager.

Comment (MartinWeindel) [reply pkg/operator/controller/garden/garden/components.go:837]: Sorry for late response, but I don't understand your statement.
If there is a `TXT` record `_acme-challenge.{dashboard,identity,...}.<ingress-domain>`, it has no consequences for `{dashboard,identity,...}.<ingress-domain>` as it is a different domain name. If there is a concrete DNS record for such a name, it will resolve normally. Otherwise it will be resolved by the wildcard record. So where is the issue?
I have even tried it out. It worked as expected. I saw some DNS caching effects, depending in which order you add the DNS records and perform DNS queries, but this is standard DNS behaviour and disappears if you wait or change the DNS server.

Comment (MartinWeindel) [reply pkg/operator/controller/garden/garden/components.go:837]: @timebertt

Comment (timebertt) [reply pkg/operator/controller/garden/garden/components.go:837]: > If there is a concrete DNS record for such a name, it will resolve normally. Otherwise it will be resolved by the wildcard record.

We expected it to behave exactly like you said. This might be true for some DNS implementations, but not for all. In our tests with Google CloudDNS, the wildcard record was unresolvable, i.e., "hidden" by the concrete records (even TXT records).

Comment (MartinWeindel) [reply pkg/operator/controller/garden/garden/components.go:837]: This is a surprising different behaviour of Google CloudDNS 🤯. BTW, when using the Gardener cert-management it is only a temporary issue, as the `TXT` records for the DNS challenges are removed as soon as the certificate request has finished.

@timebertt 
I don't think this PR is the right place to replace the wildcard DNS records for ingress. We need to discuss it and think about side effects. It should be handled in a separate PR.
